nixos/modules/security/misc.nix

```diff
@@ -136,5 +147,9 @@ with lib;
     (mkIf (config.security.virtualisation.flushL1DataCache != null) {
       boot.kernelParams = [ "kvm-intel.vmentry_l1d_flush=${config.security.virtualisation.flushL1DataCache}" ];
     })
+
+    (mkIf (!config.security.enableX32Abi) {
+      boot.kernelParams = [ "syscall.x32=n" ];
```

Comment:
Where did you find syscall.x32=n kernel param?

for comparison 32-bit ABI is handled by ia32_emulation= at https://docs.kernel.org/admin-guide/kernel-parameters.html

Glancing at the https://github.com/torvalds/linux/blob/master/arch/x86/entry/common.c I don't see an x32 equivalent.

Reply:
It looks like the parameter is added by [a Debian patch](https://salsa.debian.org/kernel-team/linux/-/blob/d0205017504bbd1fbddbc7f57c2a5531dd3bb4d7/debian/patches/features/x86/x86-make-x32-syscall-support-conditional.patch):

> Enabling x32 in the standard amd64 kernel would increase its attack surface while provide no benefit to the vast majority of its users. No-one seems interested in regularly checking for vulnerabilities specific to x32 (at least no-one with a white hat).

Given that they apparently share my security concerns, and that carrying more kernel patches has a higher cost in terms of maintenance effort and upstream drift than just toggling on a configuration knob, I’m afraid I lean a bit more towards not making this change than when I thought we could get a bootloader toggle natively. It seems the patch has required [a fair amount of (admittedly trivial) maintenance](https://salsa.debian.org/kernel-team/linux/-/commits/d0205017504bbd1fbddbc7f57c2a5531dd3bb4d7/debian/patches/features/x86/x86-make-x32-syscall-support-conditional.patch) over the years.

It seems like upstream were broadly receptive to the idea when the patch was posted to LKML, so maybe this could go upstream and make it easier for downstream distributions to toggle this on without exposing people to additional risk. As noble an art as code golf is, though, I wonder if it’s really worth it. It would be one thing if x32 had achieved any kind of real adoption in the ecosystem…

Reply:
Aww. Since that parameter is not upstream, I will close this PR.